e2e/gyp_no_install_script/snapshots/bzlmod/segfault-handler_defs.bzl

```diff
@@ -4,441 +4,280 @@
-        tags = ["manual"],
-        execution_requirements = {
+        has_lifecycle_build_target = True,
+        transitive_closure_pattern = True,
+        npm_package_target = "@@aspect_rules_js~~npm~npm__segfault-handler__1.3.0//:pkg",
+        package_store_name = "segfault-handler@1.3.0",
+        lifecycle_hooks_env = {},
+        lifecycle_hooks_execution_requirements = {
             "no-sandbox": "1",
         },
-        mnemonic = "NpmLifecycleHook",
-        progress_message = "Running lifecycle hooks on npm package segfault-handler@1.3.0",
-        env = {},
         use_default_shell_env = True,
-    )
-
-    # post-lifecycle npm_package
-    _npm_package_internal(
-        name = "{}/pkg_lc".format(store_target_name),
-        src = ":{}/lc".format(store_target_name),
-        package = "segfault-handler",
-        version = "1.3.0",
-        tags = ["manual"],
+        exclude_package_contents = [],
     )
 
 # Generated npm_package_store and npm_link_package_store targets for npm package segfault-handler@1.3.0
 # buildifier: disable=function-docstring
 def npm_link_imported_package_store(name):
-    bazel_package = native.package_name()
-    link_packages = {
-        "": ["segfault-handler"],
-    }
-    if bazel_package in link_packages:
-        link_aliases = link_packages[bazel_package]
-    else:
-        link_aliases = ["segfault-handler"]
-
-    link_alias = None
-    for _link_alias in link_aliases:
-        if name.endswith("/{}".format(_link_alias)):
-            # longest match wins
-            if not link_alias or len(_link_alias) > len(link_alias):
-                link_alias = _link_alias
-    if not link_alias:
-        msg = "name must end with one of '/{{ {link_aliases_comma_separated} }}' when called from package 'segfault-handler'; recommended value(s) are 'node_modules/{{ {link_aliases_comma_separated} }}'".format(link_aliases_comma_separated = ", ".join(link_aliases))
-        fail(msg)
-
-    link_root_name = name[:-len("/{}".format(link_alias))]
-    store_target_name = ".aspect_rules_js/{}/segfault-handler@1.3.0".format(link_root_name)
-
-    # terminal package store target to link
-    _npm_link_package_store(
-        name = name,
-        package = link_alias,
-        src = "//:{}".format(store_target_name),
-        visibility = ["//visibility:public"],
-        tags = ["manual"],
-    )
-
-    # filegroup target that provides a single file which is
-    # package directory for use in $(execpath) and $(rootpath)
-    native.filegroup(
-        name = "{}/dir".format(name),
-        srcs = [":{}".format(name)],
-        output_group = "package_directory",
-        visibility = ["//visibility:public"],
-        tags = ["manual"],
+    return _npm_link_imported_package_store(
+        name,
+        package = PACKAGE,
+        version = VERSION,
+        root_package = _ROOT_PACKAGE,
+        link_packages = {
+            "": ["segfault-handler"],
```

Comment:
This string is in a constant up top, is that easy to use here?

Reply:
I can try, I was a bit intimidated by how the data flows through here until the very end (though I also think it's simpler now 😛). I'll poke at it

Reply:
ok i did it and pushed it as a separate commit on top of this, it was ok-ish, except the serialization code is kinda annoying to hack around. wdyt?